common/meta/grpc/src/grpc_client.rs

```diff
@@ -165,6 +218,51 @@ impl MetaGrpcClient {
         Ok(client)
     }
 
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub async fn set_endpoints(&self, endpoints: Vec<String>) -> Result<()> {
+        let mut eps = self.endpoints.write().await;
+        *eps = endpoints;
+        Ok(())
+    }
+
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub async fn sync_endpoints(&self) -> Result<()> {
+        let endpoints = {
+            let eps = self.endpoints.read().await;
+            (*eps).clone()
+        };
+        let channel = self.conn_pool.get(&endpoints).await?;
+        tracing::info!("connecting with channel: {:?}", channel);
+
+        let mut client = MetaServiceClient::new(channel.clone());
+        let endpoints = client
+            .member_list(Request::new(MemberListRequest {
+                data: "".to_string(),
+            }))
+            .await?;
+        let result: Vec<String> = endpoints.into_inner().data;
+        self.set_endpoints(result).await?;
+        Ok(())
+    }
+
+    async fn auto_sync_endpoints(_client: Arc<MetaGrpcClient>, mut cancel_rx: Receiver<()>) {
+        loop {
+            select! {
+                _ = &mut cancel_rx => {
+                    return;
+                }
+                _ = tokio::time::sleep(Duration::from_secs(10)) => {
+                    // let r = client.sync_endpoints();
```

Comment:
Not used anymore?

Reply:
Comment out temporarily, will be used in the next PR